adding in mini hack guide (#44)
fhdsl/AnVIL_Demos: mini-hack.Rmd

```{r echo = FALSE}
knitr::opts_chunk$set(out.width = "100%")
```

> # (PART\*) MAGE mini-hackathon {.unnumbered}

Reviewer: ```suggestion
# (PART\*) MAGE mini-hackathon {-}
```

# Overview

> Every dataset tells a story, yet deciphering exactly how a figure was constructed can feel like solving a puzzle with missing pieces. Reproducibility is a common challenge. This mini-hackathon challenges teams to recreate figures from the MAGE RNA sequencing study using real omics data from the 1000 Genomes Project. Teams can also create additional visualizations from the open data. Through hands-on experience with cloud-based tools on AnVIL, participants will discover that computational reproducibility requires creativity, problem-solving, and detective work that go beyond simply following published protocols.

Reviewer: ```suggestion
Every dataset tells a story, yet deciphering exactly how a figure was constructed can feel like solving a puzzle with missing pieces. [Reproducibility](https://en.wikipedia.org/wiki/Reproducibility) is a common challenge. This mini-hackathon challenges teams to recreate figures from the MAGE RNA sequencing study using real omics data from the 1000 Genomes Project. Teams can also create additional visualizations from the open data. Through hands-on experience with cloud-based tools on AnVIL, participants will discover that computational reproducibility requires creativity, problem-solving, and detective work that go beyond simply following published protocols.
```

**References**:

-   Paper to reproduce here: <https://pmc.ncbi.nlm.nih.gov/articles/PMC11291278/>

-   Github repository companion for the paper here: <https://github.com/mccoy-lab/MAGE>

-   Data from the paper can be found here: <https://zenodo.org/records/10535719>

> -   Original Workspace: <https://anvil.terra.bio/#workspaces/anvil-outreach/demos-explore-mage>

Reviewer: ```suggestion
-   Original Workspace: <https://anvil.terra.bio/#workspaces/anvil-outreach/demos-mage-minihack>
```
Please update throughout :)

> 

Reviewer: Would be great to add H2s for Skills Level and Learning Objectives

example : 
https://github.com/fhdsl/AnVIL_Demos/blob/65d4a60fc5bd2fe252d4baf6f81621c49d2d2a39/03-single-cell-with-bioconductor.Rmd#L13-L32

> ## Prepare the Workspace

Reviewer: ```suggestion
## Locate the Startup Script
```

> Be sure to use the start up script when creating an RStudio Cloud Environment on AnVIL. Paste the path below into the Startup script _Optional_ box to install `AnVILGCP`, `vcfR`, and `bcftools`. The start-up script can be found in this workspace under the Data tab, in the other data option there it will be under Workspace data.

Reviewer: ```suggestion
We will use a start up script when creating an RStudio Cloud Environment on AnVIL. Copy the path below: it will help us install `AnVILGCP`, `vcfR`, and `bcftools` automatically. The start-up script can be found in the original workspace under the [Data tab](https://anvil.terra.bio/#workspaces/anvil-outreach/demos-mage-minihack/data), under "Workspace Data", or in the [workspace files](https://anvil.terra.bio/#workspaces/anvil-outreach/demos-mage-minihack/files). 
```

For example, format of the start-up script would be: 
```         

> gs://fc-7e522183-41b0-4a9b-8f48-699fe8d114e8/install-conda.sh

Reviewer: ```suggestion
gs://fc-9a141623-268f-47f8-8466-1747b5980b67/install-conda.sh
```
I think this should be the literal address to the one in this workspace

```

> Here is a video tutorial on finding the path of the startup script: <iframe src="https://support.terra.bio/hc/article_attachments/360089799392" height="500" width="800" title="Tutorial"> </iframe>.

Reviewer: why not a simpler solution directly from AnVIL under https://anvil.terra.bio/#workspaces/anvil-outreach/demos-mage-minihack/files? 

<img width="464" height="330" alt="Image" src="https://github.com/user-attachments/assets/21ab111e-17ab-4278-9eed-83124580721b" />

Author: made the change! didn't know if the video method was best practices

To read more on preconfiguring a Cloud Environment using startup scripts, click [here](https://support.terra.bio/hc/en-us/articles/360058193872-Preconfigure-a-Cloud-Environment-with-a-startup-script#h_01J5R7WP1WFDKV5H5P96883TXZ).

> # Setting up your Workspace

Reviewer: ```suggestion
# Setting up your Workspace {#mini-hack-preparation}
```

1.  Clone the AnVIL workspace: <https://anvil.terra.bio/#workspaces/anvil-outreach/demos-mage-minihack>

-   For steps on launching Terra and cloning workspaces, read [here](https://hutchdatascience.org/AnVIL_Demos/what-is-anvil-exercises.html#launch-terra).

> 2.  Then, follow [these steps](https://hutchdatascience.org/AnVIL_Demos/human-genetic-variation-in-mage-exercises.html#launch-rstudio) to launch RStudio within your AnVIL workspace. 

Reviewer: I would add a bit more info here, perhaps a screenshot showing where to paste the link to the startup script when launching the environment

Author: added a screenshot and moved around startup script instructions to make it flow better

> 3. Open `Reproducibility_in_Action.Rmd` and follow through the notebook. 

Reviewer: I'd add a bit more detail - eg they need to find this under the Data tab

  - **Ensure that you replace _add your Google Project ID here_ with your Google Project ID in line 288 of your R Markdown file.**
  - To locate this information: In the dashboard page of your workspace, you will see **Google Project ID** under the Cloud Information section. The format of it will follow 'terra-xxxxxxxx'. 

> 4. At the end of the exercise, remember to shut down compute! Refer to steps [here](https://hutchdatascience.org/AnVIL_Demos/human-genetic-variation-in-mage-wrap-up.html#shut-down-compute).  

Reviewer: I'd move this to the end of the exercises

# Exercises

This hands-on activity will help you explore [MAGE](https://www.internationalgenome.org/data-portal/data-collection/mage_rnaseq), an open-access RNA sequencing dataset of lymphoblastoid cell lines from 731 individuals from the [1000 Genomes Project](https://www.internationalgenome.org/). As part of this exploration, we will attempt to recreate various figures from this [paper](https://pmc.ncbi.nlm.nih.gov/articles/PMC11291278/).

The GitHub repository can be found [here](https://github.com/mccoy-lab/MAGE).

Processed data can be found in Zenodo [here](https://zenodo.org/records/10535719) or in Dropbox [here](##0). 

> By the end of this module, you will be able to: 

Reviewer: Move these to the overview, under Learning Objectives

- Set up and manage an R analysis environment on AnVIL using cloud-native tools
- Import, reshape, and join multiple genomic data types (expression counts, sample metadata, variant calls, and genome annotations)
- Import data into an AnVIL workspace from multiple sources
- Apply core tidyverse operations including piping, filtering, joining, and mutation
- Extract and visualize expression quantitative trait loci (eQTL) data
- Reproduce figures from a peer-reviewed publication using open-access data

> 

Reviewer: ```suggestion
:::{.notice}
To follow along with these exercises, you will need to complete the steps described in the [Preparation](#mini-hack-preparation) guide for this demo.
:::
```

8. **Visualization**

A combined violin and boxplot is produced using ggplot2, displaying log2-normalized GSTP1 counts stratified by rs115070172 genotype, closely matching the published figure.

### Recreating Figure 5D: GSTP1 Expression by Population

Building on the joined dataset, we will create a new column distinguishing Peruvian (PEL) from non-PEL samples using a conditional mutation, then produce an analogous violin/boxplot stratified by population label.

### Independent Extension: A Second eQTL

You will independently apply the full workflow to a second eQTL (rs7927381 × GSTP1), locating the SNP, subsetting the VCF, joining with expression data, and generating a comparable plot. 

### Additional Exploration

Several optional prompts are provided to you, including faceting plots by population or continental group, visualizing genotype frequency distributions, comparing GSTP1 expression by sex, and replicating Figure 1B using principal component data. 

## Learn More

```{r, echo=FALSE}

> ottrpal::include_slide("https://docs.google.com/presentation/d/16DHXD2KNyjjP2mKzPDHmNE9loYy6OgqGG0-Kn8CLFak/edit#slide=id.g369240085c8_0_136")

Reviewer: New Deck for this activity here: https://docs.google.com/presentation/d/1wVFAuh0x1Oog79Dkgsp7QPpnCwkhGTxqezebDMPt15c/edit?usp=sharing - please let me know if you need edit access!

Author: changed!

```

```{r, echo=FALSE}
ottrpal::include_slide("https://docs.google.com/presentation/d/16DHXD2KNyjjP2mKzPDHmNE9loYy6OgqGG0-Kn8CLFak/edit#slide=id.g369240085c8_0_126")
```

## Provide Feedback

::: {.notice}

> Fill out [this poll](https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=GBCC2025) to share your feedback

Reviewer: ```suggestion
Fill out [this poll](https://docs.google.com/forms/d/e/1FAIpQLScrDVb_utm55pmb_SHx-RgELTEbCCWdLea0T3IzS0Oj00GE4w/viewform?usp=pp_url&entry.1565230805=AnVIL+Demos+Mini+Hack) to share your feedback
```

## Skills Level

::: notice
*Genetics*

**Beginner**: some genetics knowledge helpful

*Programming skills*

**Beginner**: some programming experience helpful
:::

## Learning Objectives

By the end of this module, you will be able to:

-   Set up and manage an R analysis environment on AnVIL using cloud-native tools
-   Import, reshape, and join multiple genomic data types (expression counts, sample metadata, variant calls, and genome annotations)
-   Import data into an AnVIL workspace from multiple sources
-   Apply core tidyverse operations including piping, filtering, joining, and mutation
-   Extract and visualize expression quantitative trait loci (eQTL) data
-   Reproduce figures from a peer-reviewed publication using open-access data 

> **References**:

Reviewer: ```suggestion
1. Set up and manage an R analysis environment on AnVIL using cloud-native tools
 
2. Import, reshape, and join multiple genomic data types (expression counts, sample metadata, variant calls, and genome annotations)

3. Import data into an AnVIL workspace from multiple sources

4. Apply core tidyverse operations including piping, filtering, joining, and mutation

5. Extract and visualize expression quantitative trait loci (eQTL) data

6. Reproduce figures from a peer-reviewed publication using open-access data 

## References:
```
